pkg/cloudprovider/cloudprovider.go

```diff
@@ -109,22 +109,22 @@ func (c *CloudProvider) Create(ctx context.Context, nodeClaim *karpv1.NodeClaim)
 		return nil, cloudprovider.NewNodeClassNotReadyError(stderrors.New(nodeClassReady.Message))
 	}
 	if nodeClassReady.IsUnknown() {
-		return nil, fmt.Errorf("resolving NodeClass readiness, NodeClass is in Ready=Unknown, %s", nodeClassReady.Message)
+		return nil, cloudprovider.NewNodeClassNotReadyError(fmt.Errorf("resolving NodeClass readiness, NodeClass is in Ready=Unknown, %s", nodeClassReady.Message))
```

Comment:
This should be just `cloudprovider.NewCreateError` (does not neccesarily mean it is not ready)

Reply:
Changed this to be the NewCreateError.
Added Reason Constant to be more specific.